test/create_pool.test.ts

```diff
@@ -0,0 +1,224 @@
+import { ONE_ALPH, web3 } from '@alephium/web3'
+import { getSigner } from '@alephium/web3-test'
+import { PrivateKeyWallet } from '@alephium/web3-wallet'
+import { deployInvariant, expectError, expectErrorCode } from '../src/utils'
+import { CLAMMError, InvariantError, PercentageScale } from '../src/consts'
+import { getPool, initPool, initFeeTier, initTokensXY } from '../src/testUtils'
+
+web3.setCurrentNodeProvider('http://127.0.0.1:22973')
+let admin: PrivateKeyWallet
+let poolCreator: PrivateKeyWallet
+
+describe('invariant tests', () => {
+  beforeAll(async () => {
+    admin = await getSigner(ONE_ALPH * 1000n, 0)
+    poolCreator = await getSigner(ONE_ALPH * 1000n, 0)
+  })
+  test('create pool', async () => {
+    const protocolFee = 10n ** (PercentageScale - 2n)
+    const invariant = await deployInvariant(admin, protocolFee)
+
+    const fee = 5n * 10n ** (PercentageScale - 1n)
+    const tickSpacing = 100n
+    await initFeeTier(invariant, admin, fee, tickSpacing)
+
+    const supply = 10n ** 6n + 1000n
+    const [tokenX, tokenY] = await initTokensXY(admin, supply)
+
+    const initTick = 0n
+    const initSqrtPrice = 10n ** 24n
+    await initPool(
+      invariant,
+      poolCreator,
+      tokenX,
+      tokenY,
+      fee,
+      tickSpacing,
+      initSqrtPrice,
+      initTick
+    )
+
+    await getPool(invariant, tokenX, tokenY, fee, tickSpacing)
```

Comment:
the state of the pool is not checked anywhere, it is worth checking at least here to catch potential regression errors